core/tests/webdriverio_utils/BlogPages.js

```diff
@@ -0,0 +1,308 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the blog pages, for use in WebdriverIO
+ * tests.
+ */
+
+var action = require('./action.js');
+var waitFor = require('./waitFor.js');
+var general = require('./general.js');
+var BlogDashboardPage = require('./BlogDashboardPage.js');
+var forms = require('../webdriverio_utils/forms.js');
+
+
+var BlogPages = function() {
+  var blogDashboardPage = null;
+  var noResultsFoundContainer = $('.e2e-test-no-results-shown');
+  var oppiaBlogHomePageCardContainer = $('.e2e-test-oppia-blog-home-page-card');
+  var oppiaAuthorProfilePageCardContainer = $(
+    '.e2e-test-oppia-author-profile-page-card');
+  var blogHomepageWelcomeHeading = $('.e2e-test-blog-welcome-heading');
+  var oppiaAvatarImage = $('.e2e-test-oppia-avatar-image');
+  var oppiaBlogPostPageContainer = $('.e2e-test-oppia-blog-post-page-card');
+  var blogPostTitleContainer = $('.e2e-test-blog-post-page-title-container');
+  var authorProfilePhoto = $('.e2e-test-author-profile-photo');
+  var authorNameContainer = $('.e2e-test-author-name');
+  var blogPostTagFilter = $('.e2e-test-tag-filter-component');
+  var blogPostSearchField = $('.e2e-test-search-field');
+  var blogPostsList = $('.e2e-test-blog-post-list');
+  var navigateToBlogHomePageButton = $('.e2e-test-back-button');
+  var paginationNextButton = $('.e2e-test-pagination-next-button');
+  var paginationPrevButton = $('.e2e-test-pagination-prev-button');
+  var paginationContainer = $('.e2e-test-pagination');
+  var searchInput = $('.e2e-test-search-input');
+  var blogPostTilesSelector = function() {
+    return $$('.e2e-test-blog-post-tile-item');
+  };
+  var blogPostRecommendationTilesSelector = function() {
+    return $$('.e2e-test-blog-post-tile-item');
+  };
+
+  this.submitSearchQuery = async function(searchQuery) {
+    await waitFor.pageToFullyLoad();
```

Comment:
I think this should go at the end of whatever function navigated to the new page, not at the start of this function (which may or may not be the next function called)

Reply:
Done